### PR TITLE
Issue 108: Add SDO APIs to list, download, and delete public keys

### PR DESCRIPTION
**Pull request by bmpotter**
See https://github.com/open-horizon/SDO-support/issues/108

Signed-off-by: Bruce Potter <bp@us.ibm.com>

### CODE REVIEW

File: ocs-api/main.go [hunk 1]
```diff
@@ -349,55 +355,179 @@ func postVoucherHandler(w http.ResponseWriter, r *http.Request) {
...
+	// Send file names to client
+	httpCode := http.StatusOK
+	if len(pubKeyMap) == 0 {
+		httpCode = http.StatusNotFound
+	}
+	outils.WriteJsonResponse(httpCode, w, pubKeyMap)
+}
+
+//============= GET /api/orgs/{org-id}/keys/{key-name} =============
+// Reads/returns an already imported voucher
```
Comment (lorenzoking) [inline ocs-api/main.go:421]: Did you mean to say "Reads/returns an already existing public key"?

File: ocs-api/main.go [hunk 2]
```diff
@@ -349,55 +355,179 @@ func postVoucherHandler(w http.ResponseWriter, r *http.Request) {
...
+	w.Header().Set("Content-Type", "application/octet-stream")
+	w.Header().Set("Content-Disposition", "attachment; filename="+keyName)
+	http.ServeFile(w, r, pubKeyFileName)
 }
-*/
 
-//============= POST /api/keys =============
-// Receives in the body json containing the arguments required to run import-owner-key-pairs.sh script. Then it creates and import key pair into master keystore. Imports them into our keystore.
+//============= DELETE /api/orgs/{org-id}/keys/{key-name} =============
+// Reads/returns an already imported voucher
```
Comment (lorenzoking) [inline ocs-api/main.go:462]: Just the comment 
"Deletes an already imported key pair (public and private keys)" ?

**Review by lorenzoking — CHANGES_REQUESTED** [review]
Just updates to the description comment section for the new API functions that were created

File: ocs-api/main.go (hunk 1 above)
Comment (bmpotter) [reply ocs-api/main.go:421]: Yes i did, thx for catching. Fixed

File: ocs-api/main.go (hunk 2 above)
Comment (bmpotter) [reply ocs-api/main.go:462]: Thx. Fixed.

**Review by lorenzoking — APPROVED** [review]

**Pull request merged**